source/Server.Extensibility/Extensions/Infrastructure/Web/Api/OctopusJsonRegistration.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Text;
+
+namespace Octopus.Server.Extensibility.Extensions.Infrastructure.Web.Api
+{
+    public class OctopusJsonRegistration<TResource> : BaseResponseRegistration
+    {
+        public OctopusJsonRegistration(HttpStatusCode statusCode = HttpStatusCode.OK, string descriptionSuffix = "resource returned")
+            : base(statusCode, $"{FriendlyId(typeof(TResource))} {descriptionSuffix}")
+        {
+            Type = typeof(TResource);
+
+        }
+
+        public Type Type { get; }
+
+        public IOctoResponseProvider Response(TResource model) => new WrappedResponse(new OctoDataResponse(model));
+
+        static string FriendlyId(Type type, bool fullyQualified = false)
```

Comment:
Had to add the friendly ID here to get a better description for swagger. Worth keeping?

Reply:
Get from swagger